src/mlpack/methods/ann/layer/multihead_attention_impl.hpp

```diff
@@ -332,18 +334,19 @@ Gradient(const MatType& input,
   // The shape of gradient : (4 * embedDim * embedDim + 4 * embedDim, 1).
   gradient.set_size(arma::size(weights));
 
-  const CubeType q(const_cast<MatType&>(input).memptr(),
-      embedDim, tgtSeqLen, batchSize, false, false);
-  const CubeType k(const_cast<MatType&>(input).memptr() +
-      (selfAttention ? 0 : q.n_elem), embedDim, srcSeqLen, batchSize, false,
-      false);
-  const CubeType v(const_cast<MatType&>(input).memptr() +
-      (selfAttention ? 0 : (q.n_elem + k.n_elem)), embedDim, srcSeqLen,
-      batchSize, false, false);
+  CubeType q;
+  MakeAlias(q, input, embedDim, tgtSeqLen, batchSize, 0, false);
+  CubeType k;
+  MakeAlias(k, input, embedDim, srcSeqLen, batchSize,
+      (selfAttention ? 0 : q.n_elem), false);
+  CubeType v;
```

Comment:
```suggestion
  CubeType q, k, v;
  MakeAlias(q, input, embedDim, tgtSeqLen, batchSize, 0, false);
  MakeAlias(k, input, embedDim, srcSeqLen, batchSize,
      (selfAttention ? 0 : q.n_elem), false);
```
(same suggestion as above, again up to you)